transports/websocket/src/framed.rs

```diff
@@ -277,14 +163,99 @@ where
     fn address_translation(&self, server: &Multiaddr, observed: &Multiaddr) -> Option<Multiaddr> {
         self.transport.lock().address_translation(server, observed)
     }
+
+    fn poll(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<libp2p_core::transport::TransportEvent<Self::ListenerUpgrade, Self::Error>> {
+        let mut transport = self.transport.lock();
+        let inner_event = match Transport::poll(Pin::new(transport.deref_mut()), cx) {
+            Poll::Ready(ev) => ev,
+            Poll::Pending => return Poll::Pending,
+        };
+        drop(transport);
```

Comment:
A way to avoid the explicit drop is to create a smaller, inner scope but that is a style-preference :)

Reply:
Good idea; done in b19e11a.